x-pack/platform/plugins/shared/cases/public/components/system_actions/cases/cases_params.tsx

```diff
@@ -274,6 +310,22 @@ export const CasesParamsFieldsComponent: React.FunctionComponent<
           />
         </EuiFlexItem>
       </EuiFlexGroup>
+    );
+  }, [editSubActionProperty, index, reopenClosedCases]);
+
+  if (isAttackDiscoveryRuleType) {
+    return templateSelectorComponent;
+  }
+
+  return (
```

Comment:
Why break apart all of the sections into memoized sub sections? If so, more of a nit here, but I would create separate components (i.e. `<GroupByComponent {...} />`) rather than wrapping the different sections in `useMemo(() => { ... })`.

Reply:
It allows me easily to return only the `templateSelectorComponent` section in case of attack discovery rule type and makes code cleaner IMO. Otherwise, I would need to hide all the unnecessary parts within the `return` with `{!isAttackDiscoveryRuleType && (...)}`.

Regarding separate components that can be done, but I was thinking that it is better to do in a separate PR to keep the intend of this one scoped only to the needed updates, taking into account the time pressure 😄